lib/logger/lib/logger.ex

```diff
@@ -330,7 +330,7 @@ defmodule Logger do
 
   """
   def add_backend(backend, opts \\ []) do
-    _ = if opts[:flush], do: GenEvent.which_handlers(:error_logger)
+    _ = if opts[:flush], do: flush()
```

Comment:
We don't need to because the Logger process is called when the backend handler is added or removed and so all flushable messages at this point are guaranteed to be handled because they will be ahead of the add/remove handler call in the Logger process message queue. However we "flush" the `:error_logger` so that any message sent to `:error_logger` before this point have been forwarded to Logger - so that they are handled before the backend is added/removed.